File: app/models/UserWatch.java
package models;

import java.util.List;
import siena.*;

@Table("UserWatch")
public class UserWatch extends Model {

    @Id
    public Long id;

    @Column("user_id")
    @Max(200) @NotNull
    public Long userId;

    @Column("watch_id")
    @Max(200) @NotNull
    public Long watchId;

    /*
     * Instance Methods
     */
    public UserWatch(User u, Watch w) {
        this.userId = u.id;
        this.watchId = w.id;
    }

    public static UserWatch byUserAndWatch(User u, Watch w) {
        return Model.all(UserWatch.class).filter("userId", u.id).
                                          filter("watchId", w.id).get();
    }

    public static List<UserWatch> byWatch(Watch w) {
        return Model.all(UserWatch.class).filter("watchId", w.id).fetch();
    }
}

File: app/models/UserWatch.scala
package models

import reflect.BeanProperty
import siena._

@Table("UserWatch")
class UserWatch extends Model {

  @Id
  @BeanProperty
  var id: Long = -1

  @Column(Array("user_id"))
  @Max(200) @NotNull
  @BeanProperty
  var userId: Long = -1

  @Column(Array("watch_id"))
  @Max(200) @NotNull
  @BeanProperty
  var watchId: Long = -1
}

object UserWatch extends Model {
  def byUserAndWatch(u:User, w:Watch):UserWatch = {
    Model.all(classOf[UserWatch]).filter("userId", u.getId()).filter("watchId", w.id).get
  }

  def byWatch(w:Watch):java.util.List[UserWatch] = {
    Model.all(classOf[UserWatch]).filter("watchId", w.getId).fetch
  }

  def all = {
    Model.all(classOf[UserWatch])
  }
}
